store/cache/cache.go

```diff
@@ -117,6 +115,9 @@ func (ckv *CommitKVStoreCache) Get(key []byte) []byte {
 		// cache hit
 		return value
 	}
+	// in case of cache miss, we need to lock mutex
+	ckv.mtx.Lock()
+	defer ckv.mtx.Unlock()
 
```

Comment:
Do we not need the lock for the read on line 113?

`	value, ok := ckv.cache.Get(keyStr)`

Reply:
the cache is threadsafe, so it should be ok to access in this manner, I'm still thinking about the potential interactions though in the event of a write coming in at the same time that dirties the cache, so it may not be ok. This was certiks suggested fix though, but i'm still evaluating it so we shouldn't include this in the next build IMO